### PR TITLE
codec_adapter: fix local buffer prepare

### PR DESCRIPTION
**Pull request by cujomalainey**
We are only setting the parameters on the allocation of the local buffer
which means if the params change on the next run it will not change and
break things.

Signed-off-by: Curtis Malainey <cujomalainey@chromium.org>

### CODE REVIEW

**Comment by cujomalainey** [comment]
Possible fix for #3966

**Review by dbaluta — APPROVED** [review]

**Review by RDharageswari — COMMENTED** [review]
LGTM

**Review by lgirdwood — APPROVED** [review]

**Comment by cujomalainey** [comment]
Waiting for @stolx to test before putting up for full review

**Review by mrajwa — APPROVED** [review]

**Comment by cujomalainey** [comment]
Given number of LTGMs putting up for full review so once test comes back we can just merge right away if test is good

**Review by keyonjie — APPROVED** [review]
Nice catch, thanks!

**Comment by lgirdwood** [comment]
> checked issue #3966 with this fix - still can be reproduced

Sound like this is two bugs and this PR fixes one of them.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Pull request merged**
